docs/sql-ref-syntax-qry-select-transform.md

```diff
@@ -0,0 +1,262 @@
+---
+layout: global
+title: TRANSFORM
+displayTitle: TRANSFORM
+license: |
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+ 
+     http://www.apache.org/licenses/LICENSE-2.0
+ 
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+---
+
+### Description
+
+The `TRANSFORM` clause is used to specify a Hive-style transform (`SELECT TRANSFORM`/`MAP`/`REDUCE`)
+query specification to transform the input by running a specified script. Users can
+plug in their own custom mappers or reducers in the data stream by using features natively supported
+in the Spark SQL. In order to run a custom mapper script `map_script` or a custom
+reducer script `reduce_script` the user can issue the command which uses the `TRANSFORM`
+clause to embed the mapper or the reducer scripts.
+
+### Syntax
+
+```sql
+SELECT { TRANSFORM ( named_expression [ , ... ] ) | MAP named_expression [ , ... ] | REDUCE named_expression [ , ... ] }
+    [ rowFormat ]
+    [ RECORDWRITER recordWriter_class ]
+    USING script [ AS ( [ col_name [ col_type ] ] [ , ... ] ) ]
+    [ rowFormat ]
+    [ RECORDREADER recordReader_class ]
+```
+
+While `rowFormat` are defined as
+```sql
+{ ROW FORMAT SERDE serde_class [ WITH SERDEPROPERTIES serde_props ] | 
+    ROW FORMAT DELIMITED
+        [ FIELDS TERMINATED BY fields_terminated_char [ ESCAPED BY escapedBy ] ]
+        [ COLLECTION ITEMS TERMINATED BY collectionItemsTerminatedBy ]
+        [ MAP KEYS TERMINATED BY keysTerminatedBy ]
+        [ LINES TERMINATED BY linesSeparatedBy ]
+        [ NULL DEFINED AS nullDefinedAs ] }
+```
+
+### Parameters
+
+* **named_expression**
+
+    An expression with an assigned name. In general, it denotes a column expression.
+
+    **Syntax:** `expression [AS] [alias]`
+
+* **row_format**    
+
+    Spark uses the `SERDE` clause to specify a custom SerDe for one table. Otherwise, use the `DELIMITED` clause to use the native SerDe and specify the delimiter, escape character, null character and so on.
+
+* **SERDE**
+
+    Specifies a custom SerDe for one table.
+
+* **serde_class**
+
+    Specifies a fully-qualified class name of a custom SerDe.
+
+* **SERDEPROPERTIES**
+
+    A list of key-value pairs that is used to tag the SerDe definition.
+
+* **DELIMITED**
```

Comment:
This duplicates with the first paragraph of this doc

Reply:
Follow this doc https://github.com/apache/spark/blame/master/docs/sql-ref-syntax-ddl-create-table-hiveformat.md#L89-L93